### PR TITLE
Set default qengine to QNNPACK on ARM for quantization tests

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #83125
* #83115
* __->__ #83097
* #83089

The issue is that there is no FBGEMM or MKLDNN on ARM and QNNPACK is disabled by default for some reason. Not sure if this PR is the right approach, alternatives are probably:
1) keep these tests skipped
2) not disable QNNPACK by default on ARM
3) ?

### CODE REVIEW

File: test/quantization/core/test_docs.py [hunk 1]
```diff
@@ -14,6 +14,16 @@
 from torch.testing._internal.common_utils import IS_ARM64
 
 
+@contextlib.contextmanager
+def use_qnnpack_qengine():
+    prev = torch._C._get_qengine()
+    torch._C._set_qengine(2)
```
Comment (jerryzh168) [inline test/quantization/core/test_docs.py:20]: there is a public api for these I think, something like the following:
read: torch.backends.quantized.engine
write: torch.backends.quantized.engine = "qnnpack" 

you can also use https://github.com/pytorch/pytorch/blob/master/test/quantization/core/test_quantized_op.py#L521 to overwrite the quantized engine

Comment (soulitzer) [reply test/quantization/core/test_docs.py]: Cool, thanks!

**Review by jerryzh168 — APPROVED** [review]
lg, please make sure ci passes before landing

File: test/quantization/core/test_docs.py [hunk 2]
```diff
@@ -108,21 +109,23 @@ def _test_code(self, code, global_inputs=None):
...
             exec(expr, global_inputs)
 
-    @unittest.skipIf(IS_ARM64, "Not working on arm")
+
     def test_quantization_doc_ptdq(self):
         path_from_pytorch = "docs/source/quantization.rst"
         unique_identifier = "PTDQ API Example::"
-        code = self._get_code(path_from_pytorch, unique_identifier)
-        self._test_code(code)
+        with override_quantized_engine("qnnpack") if IS_ARM64 else contextlib.nullcontext():
```
Comment (albanD) [inline test/quantization/core/test_docs.py:116]: If we do this for every test in here, should we do it in the `setUp` function instead?

**Review by albanD — APPROVED** [review]
Sounds ok.
Small nit only

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**